docs/adrs/add-auto-import-embedded-registry.md

```diff
@@ -0,0 +1,24 @@
+# Easy way for auto adding images to k3s
+
+Date: 2024-10-2
+
+## Status
+
+Proposed
+
+## Context
+
+Since the feature for embedded registry, the users appeared with a question about having to manually import images, specially in edge environments.
+
+As a result, there is a need for a folder who can handle this action, where every image there will be watched by a controller (a child process that will run when the embedded registry is created) for changes or new images, this new images or new changes will be added to the containerd node registry, meaning that other nodes will have access to the image.
+
+This folder could it be the agent/images itself, but it will need a change in the way the images were loaded previously, this could it be the new controller handling the first load for k3s images.
```

Comment:
Propose something specific. If we do decide to periodically re-scan the existing images folder, how would we do that? A periodic scan would be easiest, but inotify and other APIs for realtime change detection are available. Processing image tarballs can be an io-intensive operation, how should we determine if the whole file should be imported without having to reprocess the entire contents - compare mtime+size perhaps?